pkg/server/server.go

```diff
@@ -291,7 +334,7 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 }
 
 func (h *handler) doMapping(identity *token.Identity, arn string) (string, []string, error) {
-	if roleMapping, exists := h.lowercaseRoleMap[arn]; exists {
+	if roleMapping, exists := h.config.CanonicalRoleMappings[arn]; exists {
```

Comment:
Now that the mapping logic has become more complex than a map lookup, we may want to move it out of server and into its own package.  The server/handler can use a new struct that encapsulates all the mapping logic.  This is where I would put the controller struct as well.